library/src/main/kotlin/com/connectrpc/protocols/ConnectInterceptor.kt

```diff
@@ -295,14 +291,11 @@ internal class ConnectInterceptor(
 }
 
 private fun Headers.toTrailers(): Trailers {
-    val trailers = mutableMapOf<String, MutableList<String>>()
-    for (pair in this.filter { entry -> entry.key.startsWith("trailer-") }) {
-        val key = pair.key.substringAfter("trailer-")
-        if (trailers.containsKey(key)) {
-            trailers[key]?.add(pair.value.first())
-        } else {
-            trailers[key] = mutableListOf(pair.value.first())
```

Comment:
I don't know why this used `value.first()`, but it basically meant we only ever kept the first trailer value for a particular key. So if a test case used multiple values for the same key, other after the first were lost. I looked at a couple of formulations, like using the more functional/streaming operators for collections. But this simple imperative for loop was the most concise and simplest to follow.